### ISSUE TITLE
TIME_OUT error

### ISSUE THREAD

**Issue by KekOrbit** [hmes98318/r6s-stats#20]
Hello! I have an error that I always get and cant solve. The error is that when I call any API callbacks I only get ex.: rank TIME_OUT and then the program stops. Anyway to solve this issue?

code:

`const R6 = require('r6s-stats-api');

let platform = 'PC';
let name = 'Kek_Orbit';

async function main() {
    let rank = await R6.rank(platform, name);
    console.log('rank', rank);
}
main();`

**Comment by reality-exe** [comment]
Sadly i think this repo has been abandoned, the repo owner hasn't responded to any issues since october 5th 2023

**Comment by KekOrbit** [comment]
Oh, thank you. By any chance, do you have an alternate for this? I have been searching for one for almost 1 month and the tracker network API, which could be good for me, is private

>  (for rainbow six siege)

**Comment by reality-exe** [comment]
No sadly, i've been searching too lol.

I'm pretty sure this package somehow grabs the data off of the api from tracker network

**Comment by KekOrbit** [comment]
> No sadly, i've been searching too lol.
> 
> I'm pretty sure this package somehow grabs the data off of the api from tracker network

yeah I have asked them but they said that they dont have any alternate coming from them.
I think that I have to think out something, but for now links do the work that redirect to tracker network I think if somehow I can grab something useful from this repo maybe.

**Comment by reality-exe** [comment]
Hey, so i've found a REST API that @SwiftCODA made, I had to make a small change to get the pc endpoint working again. Anyways here's my fork: https://github.com/reality-exe/R6-API

**Comment by KekOrbit** [comment]
I will check on it tomorrow, if it works for me I'll let you now thank you!

**Comment by KekOrbit** [comment]
Hello! So, how can I use it? Do you have any documentation?

**Comment by SwiftCODA** [comment]
> Hello! So, how can I use it? Do you have any documentation?

The README covers it fairly well I think but let me know if you think it could be more clear in any way. You essentially install the node packages and run the Express app (as instructed in README).

This creates HTTP endpoints on your localhost at your specified port (in the config file).

The documentation for the endpoints is missing but you can check the router.ts file to see where they are.

**Comment by SwiftCODA** [comment]
> Hey, so i've found a REST API that @SwiftCODA made, I had to make a small change to get the pc endpoint working again. Anyways here's my fork: https://github.com/reality-exe/R6-API

Just looked into the thing you fixed. I located the issue. Ideally the R6Platform value would be converted to the string that Ubisoft demands WITHIN the necessary HTTP request function. The reason for this is that their endpoints require different versions of the platform name. Sometimes the PC platform is "pc" or "uplay". Sometimes "xbox" is "xbl" or "console", etc.

I'll start a new issue and fix it up soon. I'm guessing the way you patched it on your fork doesn't repair the Xbox users though.

**Comment by reality-exe** [comment]
All I did was swap the PC side around, I didnt know that it was an issue for xbox so no

**Comment by KekOrbit** [comment]
> > Hello! So, how can I use it? Do you have any documentation?
> 
> The README covers it fairly well I think but let me know if you think it could be more clear in any way. You essentially install the node packages and run the Express app (as instructed in README).
> 
> This creates HTTP endpoints on your localhost at your specified port (in the config file).
> 
> The documentation for the endpoints is missing but you can check the router.ts file to see where they are.

Sorry i didnt see the notification, but the problem is that im not that good in javascript or typescript, so i dont know how to begin using this API, i've gone so far to start it and it created the clusters but i dont know what to do now? Can you please explain the next process please? If you do so, i would be very thankful!

**Comment by SwiftCODA** [comment]
> > > Hello! So, how can I use it? Do you have any documentation?
> > 
> > 
> > The README covers it fairly well I think but let me know if you think it could be more clear in any way. You essentially install the node packages and run the Express app (as instructed in README).
> > This creates HTTP endpoints on your localhost at your specified port (in the config file).
> > The documentation for the endpoints is missing but you can check the router.ts file to see where they are.
> 
> Sorry i didnt see the notification, but the problem is that im not that good in javascript or typescript, so i dont know how to begin using this API, i've gone so far to start it and it created the clusters but i dont know what to do now? Can you please explain the next process please? If you do so, i would be very thankful!

Once you run `npm run start`, you're good to go. To keep it going 24/7, you'll have to look into TSX process managers (I haven't used them for TypeScript before, only JavaScript, but they will act similarly to PM2 for example).

You're essentially opening a web server which is accessible through `http://127.0.0.1:{PORT}` or `http://localhost:{PORT}` (or from another machine on your network using your computer's local IP which should look like `http://192.168.0.XX:{PORT}`.

You may specify the port in the `config.json` file. Let me know if you have any other questions :)

**Comment by KekOrbitScripts** [comment]
oooohh thank you very much!

**Comment by KekOrbit** [comment]
Also, do you have any tips or apis to fetch data from ongoing matches?

**Comment by reality-exe** [comment]
> Also, do you have any tips or apis to fetch data from ongoing matches? 

If you mean capture live data from matches you're actively in you could use Overwolf

**Comment by KekOrbit** [comment]
> > Also, do you have any tips or apis to fetch data from ongoing matches? 
> 
> If you mean capture live data from matches you're actively in you could use Overwolf

Thank you!
